src/Session.cpp

```diff
@@ -1584,3 +1523,74 @@ AP4_Movie* CSession::CreateMovieAtom(CStream* stream)
   }
   return nullptr;
 }
+
+bool CSession::ExtractStreamProtectionData(PLAYLIST::CPeriod::PSSHSet& sessionPsshset,
```

Comment:
@glennguy for protection data extraction, do you think worth adding a check to allow extraction on representation with MP4 ContainerType only?

Reply:
I'm not sure this even applies for other containers... not enough knowledge on that sorry!

Reply:
i don't either, bento4 should be sufficient to stop parsing non-conforming data, shouldn't be a big problem
i was just thinking to try to limit the execution of this method when not needed
it can be considered later